### DISCUSSION TITLE
RFC Status

### DISCUSSION THREAD

**Discussion by brandonjpierce** [visgl/deck.gl#6058] [Q&A]
Hello! First off thank you so much for deck.gl, it has been able to solve problems for our team that we initially thought were completely impossible on the web. We were curious about the status of this RFC: https://github.com/visgl/deck.gl/blob/master/dev-docs/RFCs/v8.0/projection-system-expansion-rfc.md

Specifically, we are looking to project various layers into WGS84, including but not limited to: `TileLayer`, `ScatterplotLayer`, `GeoJsonLayer`, and `IconLayer`. We are aware that some of this is possible today with `OrthographicView` but we also have some services (mostly ArcGIS MapServers) that only send EPSG:4326 projections.

**Comment by ibgreen** [comment]
Are you mostly looking at earth centric WGS84 coordinates? We do have some support in math.gl for locally visualizing WGS84 coordinates (calculating an ENU transformation matrix around a specific point on the earth ellipsoid).

**Reply by ibgreen** [reply]
Apologies, didn't read the question carefully. There is general reprojection support in [`@math.gl/proj4`](https://math.gl/modules/proj4/docs). 

While reprojection is not built into the deck.gl layers, it can certainly be applied manually if an application load its data before passing it to the layer (rather than using the optional convenience of passing a URL string to the layer `data` prop).

Alterntative, perhaps the reprojection function could be called in a `getPosition` accessor?

**Reply by brandonjpierce** [reply]
Both excellent ideas, thank you @ibgreen! One issue we have however is that some agencies like the NGA (https://www.nga.mil/) actually require projections to be in WGS84 which is what stemmed my original question about the RFC status.

The NGA advisory link is unfortunately a 404 now but you can see articles surrounding it such as: https://www.esri.com/arcgis-blog/products/arcgis-solutions/defense/what-does-the-nga-web-mercator-advisory-mean-for-esri-defense-and-intelligence-users/

**Reply by ibgreen** [reply]
Understood. To answer the specific question, it is not an active RFC in the sense that any specific deck.gl contributor is currently allocating resources towards implementing it. 

There is a definitely general interest from a number of parties in this functionality, and we have a good understanding of what needs to be done. All that is needed is for someone to take on the work which so far has not happened. 

Technical guidance and support would certainly be provided for an interested contributor (such as yourself?) alternatively there are engineers who do contract work on deck.gl.

**Reply by brandonjpierce** [reply]
Appreciate the response!

**Reply by jfuehner** [reply]
@brandonjpierce I have a similar use-case trying to use EPSG:4326 with deck.

You can follow along [here](https://github.com/visgl/deck.gl/discussions/6090) . The recommendation was to use the `OrbitView` and then for `TileLayer` specify a custom `tileMatrix`.
